src/classes/RD2_OpportunityService_TEST.cls

```diff
@@ -924,6 +924,120 @@ private with sharing class RD2_OpportunityService_TEST {
         assertPaymentIsMatchingOpp(oppById);
     }
 
+    /***
+    * @description Verifies if Account is not getting removed in opportunity
+    * when recurring donation record with only contact is updated.
+    */
+    @isTest
+    private static void shouldNotRemoveAccountInOppWhenRDWithNoAccountIsModified() {
+        RD2_EnablementService_TEST.setRecurringDonations2Enabled();
+
+        setUpConfiguration();
+
+
+        npe03__Recurring_Donation__c rd = createRecurringDonation();
+
+        //create opp from rd with contact only
+        Opportunity opp = getOpportunityBuilderWithNoAccount(rd)
+            .withCloseDate(CLOSE_DATE)
+            .withOpenStage()
+            .build();
+        insert opp;
+
+
+        //Verify if accountid is populated in opp when only contact is provided in RD
+        List<Opportunity> opps = oppGateway.getRecords(rd);
+        System.assertEquals(1, opps.size(), 'One Installment Opp should be created on RD insert: ' + opps);
+        System.assert(opps[0].AccountId != null);
```

Comment:
The expected and actual value should be specified as well as the message of what is expected. For example: `System.assertNotEquals(null, opps[0].AccountId, 'Opp Account should be populated on the Opp insert');`